torchdp/compute_dp-sdg_privacy.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
+"""
+Both `apply_dp_sgd_analysis()` and `get_privacy_spent()` functions
+ are based on Google's TF Privacy:
+ https://github.com/tensorflow/privacy/blob/master/tensorflow_privacy/privacy/analysis/compute_dp_sgd_privacy.py
+
+Command-line script for computing privacy of a model trained with DP-SGD.
+The script applies the RDP accountant to estimate privacy budget of an iterated
+ Sampled Gaussian Mechanism.
+
+Example:
+  compute_dp_sgd_privacy
+    --dataset-size=60000 \
+    --batch-size=256 \
+    --sigma=1.12 \
+    --epochs=60 \
+    --delta=1e-5
+The output states that DP-SGD with these parameters satisfies (2.92, 1e-5)-DP.
+"""
+import argparse
+import math
+from privacy_analysis import compute_rdp, get_privacy_spent
+
+
+def apply_dp_sgd_analysis(q, sigma, steps, orders, delta):
+    """Compute and print results of DP-SGD analysis.
+
+   Compute_rdp requires that sigma be the ratio of the standard deviation of
+   the Gaussian noise to the l2-sensitivity of the function to which it is
+   added. Hence, sigma here corresponds to the `noise_multiplier` parameter
+   in the DP-SGD implementation found in privacy.optimizers.dp_optimizer
+  """
+    rdp = compute_rdp(q, sigma, steps, orders)
+
+    #  Slight adaptation from TF version, in which
+    # `get_privacy_spent()` has one more arguments and one more element in
+    #  returned tuple, because it can also compute delta for a given epsilon
+    #  (and not only compute epsilon for a targeted delta).
+    eps, opt_order = get_privacy_spent(orders, rdp, delta=delta)
+
+    print(
```

Comment:
Can you use f string here

Reply:
Sure, I prefer f strings (but here I just copied TF code as is). 
[Lines 41 to 45 to be replaced, not only 41, I don't manage to select them, sorry]
```suggestion
    print(f"DP-SGD with\n\tsampling rate = {100 * q:.3g}% and\n\tnoise_multiplier = {sigma}"
          f"\n\titerated over {steps} steps\n satisfies",
          end=" ",
    )
```